pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -607,6 +607,7 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
       int numUnavailableSegments = unavailableSegments.size();
       requestContext.setNumUnavailableSegments(numUnavailableSegments);
 
+      boolean isPartialResult = false;
```

Comment:
what is the "partial result" definition? for example if grouplimits reached for order by queries do we consider that as partial results?

i guess the question is what's the border between
1. result is correct
2. result is correct but we didn't look at the entire table (e.g. `select * limit 1000`)
3. result is not correct but each row is correct (e.g. `select ... group by key order by COUNT(*)`)
4. result is not correct, some rows might be correct some rows might not due to data size limit (`select ... join ... group by ...` when join hits maxRow limit, grouping result might be partial)
5. result is incorrect (we shouldn't return in this case)

Reply:
@walterddr my idea with this flag is to generally cover all cases where we do sent a response back (no failures), but we also know either some segments / servers were not queried (segment unavailability / server down etc) or some rows / groups were ignored.

1)  result is correct (this is no flag)

2) result is correct but we didn't look at the entire table (e.g. select * limit 1000) -> I don't think this should be flagged as partial result since the result is correct, we don't have to look the entire table up

3) result is not correct but each row is correct (e.g. `select ... group by key order by COUNT(*)`) -> I'm not sure if there is a way of knowing this in the current server -> broker response? If yes, then I do think we should flag this a partial result. I may not have full context, but is it the trimming of cross segment groups to `max(limit * 5, DEFAULT_MIN_NUM_GROUPS)` that you're referring to?

4) result is not correct, some rows might be correct some rows might not due to data size limit (select ... join ... group by ... when join hits maxRow limit, grouping result might be partial) -> Yes this should classify as partialResponse imo

5) result is incorrect (we shouldn't return in this case) -> What do you mean by incorrect here?

Reply:
I think general definition (and this is what we did in the current engine IIRC) is that partialResults flag would be set to true in either of the following:

- Result is incomplete (determined by server based on things like GROUP BY heuristics, threshold etc)
- Not all servers queried responded (determine by broker)

Basically when we are able to determine that there was more data to be processed but we couldn't either because of our own algorithms (heuristics / thresholds etc) or failed to hear back from servers.

I think (1) can also be determined by broker during merge. 

For SELECT with ORDER BY, I don't think we ever set `partialResults` on server and there is no need imo.

For SELECT without ORDER BY, servers do early termination based on LIMIT N, but again partialResults is not set and there is no need.